### PR TITLE
Fix users getting duplicate Drupal IDs.

### PR DESCRIPTION
**Pull request by DFurnes**
#### What's this PR do?
This PR fixes an issue where users with duplicate `drupal_id`s could be created (for example, when changing an email on an account in Phoenix) because we weren't using those fields to resolve and update existing accounts when POSTing to `users/`.

It also includes two clean-up commits: fixing an incorrect docblock (365af08), and removing the "key" alias to the renamed "scope" middleware (365af08).

References #334.

#### Before deploying this...
We'll want to manually clean up any duplicate `drupal_id`s before merging this, since otherwise those accounts would become "unwritable" since validation would always fail. ~~Probably a good idea to repurpose @chloealee's dedupe script?~~

__Update:__ I updated the dedupe script in 2e67a77, and tested it on a few fake "dupes" I had made before applying these fixes. Should be safe to run again on QA/production once this merges!

#### How should this be reviewed?
I'd recommend skimming the whole diff, since I ended up changing my approach a bit partway through. The changes made in 365af08 add a bit of noise, but otherwise it's all pretty straightforward.

#### Checklist
- [ ] Documentation added for changed endpoints.
- [x] Tests added for new features/bug fixes.

---
For review: @angaither @weerd @chloealee

### CODE REVIEW

**Comment by angaither** [comment]
👍

File: app/Console/Commands/RemoveDuplicateUsersCommand.php [hunk 1]
```diff
@@ -120,6 +120,49 @@ public function fire()
...
         $this->deduplicate($mobile_duplicates, 'mobile');
+
+        // Find all duplicate users by Drupal ID.
+        $drupal_duplicates = User::raw(function ($collection) {
+            return $collection->aggregate(
+                [
+                    [
+                        '$match' => [
+                            'drupal_id' => [
+                                '$ne' => '',
```
Comment (weerd) [inline app/Console/Commands/RemoveDuplicateUsersCommand.php]: what's a `ne`?

Comment (DFurnes) [inline app/Console/Commands/RemoveDuplicateUsersCommand.php]: I _think_ "not equal"? I'm not gonna lie, I just copy-pasted from the query Chloe had used above. 😇

**Comment by weerd** [comment]
Coolio!
:sheep: it!

File: app/Console/Commands/RemoveDuplicateUsersCommand.php (hunk 1 above)
Comment (weerd) [inline app/Console/Commands/RemoveDuplicateUsersCommand.php]: 👌

**Pull request merged**